pkg/validating-webhooks/hive/v1/clusterdeployment_validating_admission_hook.go

```diff
@@ -319,6 +319,11 @@ func (a *ClusterDeploymentValidatingAdmissionHook) validateCreate(admissionSpec
 	}
 
 	allErrs = append(allErrs, validateClusterPlatform(specPath.Child("platform"), cd.Spec.Platform)...)
+
+	if cd.Spec.Platform.Azure != nil && cd.Spec.ManageDNS && cd.Spec.Platform.Azure.BaseDomainResourceGroupName == "" {
```

Comment:
I'm curious why you moved this out of `validateClusterPlatform`?

Oh, is it because we don't currently have access to `cd.Spec.ManageDNS` there?

I guess the cleanest alternative is making `validateClusterPlatform` accept the `cd` instead of `cd.Spec.Platform`.

I can't decide whether it's worth doing that for the sake of keeping ~all~ more of the platform validation together in one spot.

This file isn't beautifully organized in the first place...